core/unix/signal.c

```diff
@@ -8339,15 +8371,20 @@ handle_suspend_signal(dcontext_t *dcontext, kernel_siginfo_t *siginfo,
     sig_full_cxt_t sc_full;
     ASSERT(ostd != NULL);
 
-    /* Distinguish up front from a synchronous app signal by looking for
-     * si_code where <=0 means user-sent.
-     * We distinguish further below from the rare case of an app sending
-     * SUSPEND_SIGNAL asynchronously by looking for our particular state settings
-     * that correspond to the use of this signal by DR..
-     */
-    if (siginfo->si_code > 0 || siginfo->si_signo != SUSPEND_SIGNAL)
+    if (!is_signal_for_us(siginfo, SUSPEND_SIGNAL))
         return true; /* pass to app */
 
+    if (is_sigqueue_supported() && SUSPEND_SIGNAL == NUDGESIG_SIGNUM) {
+        nudge_arg_t *arg = (nudge_arg_t *)siginfo;
```

Comment:
Should we further check is_signal_for_us(siginfo, NUDGESIG_SIGNUM)?

Reply:
But if they're the same number we've already called that.